lexibank_satterthwaitetb.py

```diff
@@ -1,192 +1,261 @@
+            # elements with superflouos information, such as the page number or
+            # footnotes. Given that they are not found in all pages
+            # in the same number when found, once more we need a manual
+            # correction. If the the first cell of the page is not listed
+            # in the `DROP_FINAL` exceptions, we default to removing only
+            # the last element
+            page_cells = page_cells[: -DROP_FINAL.get(page_cells[0], 1)]
+
+            # Once more due the PDF correction, in some cases the contents
+            # are shifted one entry to left due to empty cells. As the number
+            # of such empty cells changes from page to page, we need to
+            # test the rows, by index, to make sure the data is there.
+            page_rows = {}
+            for row_name, indexes in ROW_IDX.items():
+                if not indexes[0]:  # concept, first one
+                    idx2 = page_cells.index(indexes[1])
+                    row = page_cells[:idx2]
+                elif not indexes[1]:  # Zaozou, last one
+                    idx1 = page_cells.index(indexes[0])
+                    row = page_cells[idx1:]
                 else:
-                    row = cells[cells.index(indexes[0]): cells.index(indexes[1])]
+                    idx1 = page_cells.index(indexes[0])
+                    idx2 = page_cells.index(indexes[1])
+                    row = page_cells[idx1:idx2]
 
-                # if it is not the concept row, exclude the first element of
-                # each row (it is the language name, we already have it)
+                # In all cases except the concept row, exclude the first
+                # element (it is the language name, we already have it)
                 if row_name is not "concepts":
-                    rows[row_name] = row[1:]
+                    page_rows[row_name] = row[1:]
                 else:
-                    rows[row_name] = row
+                    page_rows[row_name] = row
 
-            page_data.append(rows)
+            # Finally collect the page data
+            page_data.append(page_rows)
 
-        with self.cldf as ds:
-            ds.add_sources()
-            ds.add_languages(id_factory=lambda l: slug(l["Name"]))
-            ds.add_concepts(id_factory=lambda c: slug(c.label))
+        return page_data
 
-            for page in pb(page_data):
-                # for each concept...
-                for idx, english in enumerate(page["concepts"]):
-                    for lang in page.keys():
-                        if lang in ["concepts", "Hanzi"]:
+    def cmd_makecldf(self, args):
+        # read page data, solving all PDF conversion problems (forms still
+        # need to be cleaned)
+        page_data = self._read_page_data()
+
+        # write the bibliographic sources
+        args.writer.add_sources()
+
+        # add the languages from the language list
```

Comment:
Please don't make ids by slugging, but maybe you're not there yet, right? But There should be a list with languages, CamelCase, in `etc/languages.tsv`.